tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs

```diff
@@ -86,27 +108,16 @@ private LabResponse GetLabResponseFromAPI(UserQueryParameters query)
                 queryDict.Add("b2cProvider", "google");
             }
 
-            UriBuilder uriBuilder = new UriBuilder("http://api.msidlab.com/api/user");
-            uriBuilder.Query = string.Join("&", queryDict.Select(x => x.Key + "=" + x.Value.ToString()));
-
-            //Fetch user
-            string result = webClient.GetStringAsync(uriBuilder.ToString()).GetAwaiter().GetResult();
-
-            if (string.IsNullOrWhiteSpace(result))
+            if (query.UserContains != null)
             {
-                throw new LabUserNotFoundException(query, "No lab user with specified parameters exists");
+                queryDict.Add("usercontains", query.UserContains);
+                queryDict.Add("AppName", query.AppName);
             }
 
-            LabResponse response = JsonConvert.DeserializeObject<LabResponse>(result);
-
-            LabUser user = response.User;
-
-            user = JsonConvert.DeserializeObject<LabUser>(result);
-
-            if (!string.IsNullOrEmpty(user.HomeTenantId) && !string.IsNullOrEmpty(user.HomeUPN))
-                user.InitializeHomeUser();
-
-            return response;
+            UriBuilder uriBuilder = new UriBuilder("http://api.msidlab.com/api/userBeta");
```

Comment:
I don't think we should use the BETA endpoint for all our users, but maybe only for the 1 user which is not in "prod"? I am mindful that BETA is more unstable and this change might hurt the reliability of our tests.

Reply:
The Beta and release end-points are in lock down until the new year. I think both are stable for our usage (at least data doesn't indicate otherwise so far). However generally a very fair point.

---
In reply to: [241009713](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/721#discussion_r241009713) [](ancestors = 241009713)

Reply:
Reason i'm using beta here is because the MSA user is not available until January when they push it live. But that depends if we want this merged sooner, or can wait. Otherwise, we usually use  the release endpoint.

---
In reply to: [241237242](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/721#discussion_r241237242) [](ancestors = 241237242,241009713)

Reply:
I'm fine to use beta for now. I suppose Bogdan suggest to limit the usage of the beta end-point for the MSA user and keep it the non MSA users to the std. end-point

---
In reply to: [241241150](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/721#discussion_r241241150) [](ancestors = 241241150,241237242,241009713)

Reply:
we can do that.

---
In reply to: [241243007](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/721#discussion_r241243007) [](ancestors = 241243007,241241150,241237242,241009713)

Reply:
using beta only for msa user now.

---
In reply to: [241243335](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/721#discussion_r241243335) [](ancestors = 241243335,241243007,241241150,241237242,241009713)